File: member-service/src/main/java/com/hedvig/memberservice/web/dto/APIErrorDTO.java
package com.hedvig.memberservice.web.dto;

import com.fasterxml.jackson.annotation.JsonTypeInfo;
import com.fasterxml.jackson.annotation.JsonTypeName;
import lombok.Value;
import org.springframework.http.HttpStatus;

@Value
@JsonTypeName("apiError")
@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
public class APIErrorDTO {
    HttpStatus status;
    String code;
    String message;
}

File: member-service/src/main/java/com/hedvig/memberservice/web/dto/APIErrorDTO.kt
package com.hedvig.memberservice.web.dto

import com.fasterxml.jackson.annotation.JsonTypeInfo
import com.fasterxml.jackson.annotation.JsonTypeName
import org.springframework.http.HttpStatus

@JsonTypeName("apiError")
@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
data class APIErrorDTO(
    val status: HttpStatus,
    val code: String,
    val message: String
)
